aten/src/ATen/native/cuda/CUDAUnaryOps.cpp

```diff
@@ -94,4 +97,19 @@ IMPLEMENT_UNARY_OP_PREQUEL(tan)
 IMPLEMENT_UNARY_OP_PREQUEL(tanh)
 IMPLEMENT_UNARY_OP_PREQUEL(trunc)
 
+#define IMPLEMENT_UNARY_OP_VEC_CUDA(op)                          \
```

Comment:
I think this part can and should be shared between CPU and CUDA. BinaryOps.cpp has examples of this. It might mean that you won't be able to use the existing IMPLEMENT_UNARY_XXX macros.